reposerver/repository/repository.go

```diff
@@ -1354,10 +1357,11 @@ func runConfigManagementPluginSidecars(ctx context.Context, appPath, repoPath st
 	}
 
 	// generate manifests using commands provided in plugin config file in detected cmp-server sidecar
-	env, err := getPluginEnvs(envVars, q, creds)
+	cleanup, env, err := getPluginEnvs(envVars, q, creds)
```

Comment:
IMO it's worth calling `cleanup()` regardless of `error` ?

Reply:
That's a good observation. Technically, with the code as written, I think it should already be called, because [`getPluginEnvs` calls `cleanup`](https://github.com/argoproj/argo-cd/pull/8389/files#diff-66db997cf6001ce63d5c5803380962f4b474aae67abacc9e86192673080d2c59R1323) in the event that it _does_ return an `error` where `cleanup` may need to be called.

I'm partial to coding with that convention (assume you'd only need to `defer cleanup()` if the returned error is non-nil). This kind of cleanup pattern matches what I've experienced in go standard libraries, e.g. [http.Do](https://pkg.go.dev/net/http#Client.Do), which states:
> On error, any Response can be ignored. A non-nil Response with a non-nil error only occurs when CheckRedirect fails, and even then the returned Response.Body is already closed.

All that being said, I'm a first time contributor here so some guidance on how to proceed would be appreciated; if you're not swayed by this explanation then I'm happy to make the change.